wallet/src/ui/app/components/transactions-card/index.tsx

```diff
@@ -3,64 +3,105 @@
 
 import cl from 'classnames';
 import { memo } from 'react';
+import { Link } from 'react-router-dom';
 
-import ExplorerLink from '_components/explorer-link';
-import { ExplorerLinkType } from '_components/explorer-link/ExplorerLinkType';
+import Icon, { SuiIcons } from '_components/icon';
 import { useMiddleEllipsis } from '_hooks';
 
 import type { TxResultState } from '_redux/slices/txresults';
 
 import st from './TransactionsCard.module.scss';
 
+const TRUNCATE_MAX_LENGTH = 8;
+const TRUNCATE_PREFIX_LENGTH = 4;
+
 function TransactionCard({ txn }: { txn: TxResultState }) {
-    const toAddrStr = useMiddleEllipsis(txn.To || '', 20);
+    const toAddrStr = useMiddleEllipsis(
+        txn.To || '',
+        TRUNCATE_MAX_LENGTH,
+        TRUNCATE_PREFIX_LENGTH
+    );
+    const fromAddrStr = useMiddleEllipsis(
+        txn.From || '',
+        TRUNCATE_MAX_LENGTH,
+        TRUNCATE_PREFIX_LENGTH
+    );
+
+    const transferStatus = txn.status === 'success' ? 'Checkmark' : 'Close';
+    const TxIcon = txn.isSender ? SuiIcons.ArrowLeft : SuiIcons.Buy;
+    const iconClassName = txn.isSender
+        ? cl(st.arrowActionIcon, st.angledArrow)
+        : cl(st.arrowActionIcon, st.buyIcon);
+
+    const date = txn?.timestamp_ms
+        ? new Date(txn.timestamp_ms).toLocaleDateString('en-us', {
```

Comment:
If you make the date formatting helper not a hook, you could leverage it here.

Reply:
Done.